Formula/go-boring.rb

```diff
@@ -0,0 +1,69 @@
+class GoBoring < Formula
+  desc "Go programming environment with boringcrypto"
+  homepage "https://go.googlesource.com/go/+/dev.boringcrypto/README.boringcrypto.md"
+  url "https://go-boringcrypto.storage.googleapis.com/go1.16.2b7.src.tar.gz"
+  sha256 "8991c28e30bdd9b34076c2871ab69aac7f306dba08d202aa7301f3adb54ad9f3"
+  license "BSD-3-Clause"
+
+  # Don't update this unless this version cannot bootstrap the new version.
+  resource "gobootstrap" do
+    on_macos do
+      if Hardware::CPU.arm?
+        url "https://storage.googleapis.com/golang/go1.16.darwin-arm64.tar.gz"
+        version "1.16"
+        sha256 "4dac57c00168d30bbd02d95131d5de9ca88e04f2c5a29a404576f30ae9b54810"
+      else
+        url "https://storage.googleapis.com/golang/go1.16.darwin-amd64.tar.gz"
+        version "1.16"
+        sha256 "6000a9522975d116bf76044967d7e69e04e982e9625330d9a539a8b45395f9a8"
+      end
+    end
+
+    on_linux do
+      url "https://storage.googleapis.com/golang/go1.16.linux-amd64.tar.gz"
+      version "1.16"
+      sha256 "013a489ebb3e24ef3d915abe5b94c3286c070dfe0818d5bca8108f1d6e8440d2"
+    end
+  end
+
+  def install
+    (buildpath/"gobootstrap").install resource("gobootstrap")
+    ENV["GOROOT_BOOTSTRAP"] = buildpath/"gobootstrap"
+
+    cd "src" do
+      ENV["GOROOT_FINAL"] = libexec
+      system "./make.bash", "--no-clean"
+    end
+
+    (buildpath/"pkg/obj").rmtree
+    rm_rf "gobootstrap" # Bootstrap not required beyond compile.
+    libexec.install Dir["*"]
+    bin.install_symlink Dir[libexec/"bin/go*"]
+
+    system bin/"go", "install", "-race", "std"
+
+    # Remove useless files.
+    # Breaks patchelf because folder contains weird debug/test files
+    rm_rf Dir[libexec/"src/debug/elf/testdata"]
+  end
+
+  test do
+    (testpath/"hello.go").write <<~EOS
+      package main
+
+      import "fmt"
```

Comment:
```suggestion
      import (
          "fmt"
          _ "crypto/tls/fipsonly"
      )
```
Let's check that this does, in fact, use a FIPS-compliant TLS configuration. Source: https://stackoverflow.com/a/57595241/10153385